check_state.pm

```diff
@@ -1178,22 +1161,256 @@ sub ldrtno {
+        open (TF, "<$tfile");
+        $tnum = <TF>;
+        close TF;
+      }
+      $tnum--;
+      if ($tnum == 0) {
+        unlink $afile;
+      }
+      if ($tnum > 0) {
+        open (TF, ">$tfile");
+        print TF $tnum;
+        close TF;
+      }
+    }
+  } # if ($val > 4) {
+  if ($val <= 4) {
+    $color = $RED;
+    my $tnum = 0;  # but, wait a little while before waking people up
+    if (-s $tfile) {
+      open (TF, "<$tfile");
+      $tnum = <TF>;
+      close TF;
+    }
+    $tnum++;
+    if ($tnum == 10) {   # How many cycles? Dan suggested 10 frames
+      send_p05vbvl_alert($val);
+    }
+    if ($tnum <= 10) {
+      open (TF, ">$tfile");
+      print TF $tnum;
+      close TF;
+    }
+  } #if ($val <= 4) {
+  return $color;
+}
+
+
+sub p24vbvl {
+  my $val = $_[0];
+  my $afile = ".p24vbvlalert";
+  my $tfile = ".p24vbvlwait";
+  $color = $BLU;
+  if ($val > 23) {
+    $color = $GRN;
+    if (-s $afile) {
+      my $tnum = 3;  # but, wait a little while before deleting lock
+      if (-s $tfile) {
+        open (TF, "<$tfile");
+        $tnum = <TF>;
+        close TF;
+      }
+      $tnum--;
+      if ($tnum == 0) {
+        unlink $afile;
+      }
+      if ($tnum > 0) {
+        open (TF, ">$tfile");
+        print TF $tnum;
+        close TF;
+      }
+    }
+  } # if ($val > 23) {
+  if ($val <= 23) {
+    $color = $RED;
+    my $tnum = 0;  # but, wait a little while before waking people up
+    if (-s $tfile) {
+      open (TF, "<$tfile");
+      $tnum = <TF>;
+      close TF;
+    }
+    $tnum++;
+    if ($tnum == 10) {   # How many cycles? Dan suggested 10 frames
+      send_p24vbvl_alert($val);
+    }
+    if ($tnum <= 10) {
+      open (TF, ">$tfile");
+      print TF $tnum;
+      close TF;
+    }
+  } #if ($val <= 23) {
+  return $color;
+}
+
+
+sub prbscr {
+  my $val = $_[0];
+  my $afile = ".prbscralert";
+  my $tfile = ".prbscrwait";
+  $color = $BLU;
+  if ($val < 2.0 && $val > -1.3) {
+    $color = $GRN;
+    if (-s $afile) {
+      my $tnum = 3;  # but, wait a little while before deleting lock
+      if (-s $tfile) {
+        open (TF, "<$tfile");
+        $tnum = <TF>;
+        close TF;
+      }
+      $tnum--;
+      if ($tnum == 0) {
+        unlink $afile;
+      }
+      if ($tnum > 0) {
+        open (TF, ">$tfile");
+        print TF $tnum;
+        close TF;
+      }
+    }
+  } # if ($val < 2.0 && $val > -1.3) {
+  if ($val >= 2 || $val <= -1.3) {
+    $color = $RED;
+    my $tnum = 0;  # but, wait a little while before waking people up
+    if (-s $tfile) {
+      open (TF, "<$tfile");
+      $tnum = <TF>;
+      close TF;
+    }
+    $tnum++;
+    if ($tnum == 3) {   # How many cycles?
+      send_prbscr_alert($val);
+    }
+    if ($tnum <= 10) {
+      open (TF, ">$tfile");
+      print TF $tnum;
+      close TF;
+    }
+  } #if ($val < 2.0 && $val > -1.3) {
+  return $color;
+}
+
+
 sub send_tank_red {
   my $obstime = ${$hash{PMTANKP}}[0];
   if (! time_curr($obstime)) {
```

Comment:
these two lines are commented out in check_state_test.pm, what's the reason for the difference here?

Reply:
They are commented in check_state_test.pm so that we can test with old telemetry.

Reply:
So they will be un-commented when we the changes are made?

Reply:
In check_state_test we always had these lines commented, I think? The idea was to be able to test on ``fabricated`` telemetry. In check_state.pm these lines are un-commented so that the alert checking happens only when real time data is flowing. In check_state_sim.pm (to be added for the SIMs) these lines will be un-commented, TEST in the subject and body of the messages will be replaced with SIM, and the mailing lists will be the same as in check_state.pm. Does is make sense?

Reply:
I've been using check_state_test.pm anytime we've had an anomaly and during recovery didn't need to send out alerts. check_state_no_alerts.pm is written in a way that still sends out alerts - I'm not sure what the original intention behind this was. It's fine for test to have the lines uncommented, will it still function if set during an anomaly? Or will we have to have a different way of not sending out alerts?